Docker build fails at npm build

I cloned the repository, without any changes I tried to build the docker image as specified in the README (`docker build -t simple-secret-santa .`).
The last step `npm run build` fails with the following error message:

```
Step 10/15 : run npm run build
 ---> Running in 2a1e80df7a6d

> simple-secret-santa@0.1.0 build
> vue-tsc --noEmit && vite build

src/components/steps/Result.vue(14,35): error TS2304: Cannot find name 'props'.
src/components/steps/Confirm.vue(28,31): error TS2304: Cannot find name 'props'.
src/components/steps/Confirm.vue(36,17): error TS2304: Cannot find name 'handleConfirm'.
src/components/steps/Confirm.vue(43,17): error TS2304: Cannot find name 'handleDismiss'.
src/components/Steps.vue(48,17): error TS2304: Cannot find name 'shouldDisplayCode'.
src/components/Steps.vue(48,47): error TS2304: Cannot find name 'handleEmitUser'.
src/components/Steps.vue(48,71): error TS2552: Cannot find name 'draws'. Did you mean 'Draw'?
src/components/Steps.vue(49,20): error TS2304: Cannot find name 'shouldDisplayConfirmation'.
src/components/Steps.vue(49,61): error TS2304: Cannot find name 'handleEmitConfirm'.
src/components/Steps.vue(49,87): error TS2304: Cannot find name 'userFrom'.
src/components/Steps.vue(50,19): error TS2304: Cannot find name 'shouldDisplayResult'.
src/components/Steps.vue(50,47): error TS2304: Cannot find name 'userTo'.
src/components/steps/Code.vue(76,19): error TS2304: Cannot find name 'handleCodeKeyup'.
src/components/steps/Code.vue(77,17): error TS2304: Cannot find name 'handleCodeFocus'.
src/components/steps/Code.vue(78,17): error TS2304: Cannot find name 'handleCodeChange'.
npm notice
npm notice New major version of npm available! 7.15.1 -> 8.19.2
npm notice Changelog: <https://github.com/npm/cli/releases/tag/v8.19.2>
npm notice Run `npm install -g npm@8.19.2` to update!
npm notice
The command '/bin/sh -c npm run build' returned a non-zero code: 2
```

Have you any idea what could cause this issue? I tried it on an Ubuntu 20.04 and a Windows 10 installation of Docker, so I think that there might be some dependencies missing / not commited. Or maybe even outdated dependencies.

Possibly it is also caused by the following warnings that come up at the `npm install` step:

```
npm WARN deprecated @volar/html2pug@0.29.8: WARNING: This project has been renamed to @johnsoncodehk/html2pug. Install using @johnsoncodehk/html2pug instead.
npm WARN deprecated vscode-pug-languageservice@0.29.8: WARNING: This project has been renamed to @volar/pug-language-service. Install using @volar/pug-language-service instead.
npm WARN deprecated vscode-typescript-languageservice@0.29.8: WARNING: This project has been renamed to @volar/typescript-language-service. Install using @volar/typescript-language-service instead.
npm WARN deprecated vscode-vue-languageservice@0.29.8: WARNING: This project has been renamed to @volar/vue-language-service. Install using @volar/vue-language-service instead.
```

--- a/package.json
+++ b/package.json
@@ -8,14 +8,18 @@
     "draw": "set -ex && $(which pip3 || which pip) install --user -r scripts/requirements.txt && $(which python3 || which python) scripts/draw.py > pairs.json"
   },
   "dependencies": {
-    "three": "^0.135.0",
-    "vue": "^3.2.25"
+    "three": "^0.145.0",
+    "vue": "^3.2.38"
   },
   "devDependencies": {
-    "@types/three": "^0.135.0",
-    "@vitejs/plugin-vue": "^2.0.1",
-    "typescript": "^4.5.4",
-    "vite": "^2.7.2",
-    "vue-tsc": "^0.29.8"
+    "@types/node": "^16.11.56",
+    "@types/three": "^0.144.0",
+    "@vitejs/plugin-vue": "^3.0.3",
+    "@vitejs/plugin-vue-jsx": "^2.0.1",
+    "@vue/tsconfig": "^0.1.3",
+    "npm-run-all": "^4.1.5",
+    "typescript": "~4.7.4",
+    "vite": "^3.0.9",
+    "vue-tsc": "^0.40.7"
   }
 }

